### ISSUE TITLE
The difference in performance between the reported results in the paper and the actual implementation of the GA3C-CADRL approach in this repo

### ISSUE THREAD

**Issue by zhangqq21** [mit-acl/gym-collision-avoidance#23]
Hello, I encountered an issue while running the code from this repository. When testing cases with 8 agents navigating randomly, I found that the success rate of the CADRL method is 83%, which matches the results in the paper. However, the success rate of GA3C-CADRL is only 39%, which does not align with the paper, where the reported success rate is over 90%. Could you please explain the reason for this? I am using the neural network model parameters provided in the repository.

**Comment by mfe7** [comment]
I have been trying to track down the exact model files used in the paper experiments. For the journal paper, here is the [wandb project](https://wandb.ai/mfe7/ga3c_cadrl/overview) that I used at the time with all the runs that were recorded. I just went back and tagged many of the runs this morning with my best guess (which is unfortunately about 5 years out of date).

There are 5 runs with the tag `access_rnn_phase2?` that seem to represent the last phase of the 5 repeated trials, each differing only in random seed, for training the LSTM model presented in the 2021 IEEE Access paper. In those runs, under the Files tab (on the left panel), under checkpoints directory, there are a list of saved model checkpoints from that run -- I presume I would have used the last one, `network_01990000.*`.

I expect that the 5 versions of that checkpoint would produce high success rate consistent with the numbers reported in the table. Also seen on the wandb page is the rolling reward for those runs, which were all ~0.9, although the score during training will be slightly different from during evaluation (for example, I remember we had larger dt during training to speed things up, the model samples from the policy's action distribution during training but uses argmax during evaluation). If the numbers on the test cases from the 5 runs described above are also much lower than the table reports, that suggests something important changed in the gym environment between the paper and now, which may be easier to debug than finding old model parameter files.

All that being said, I still believe the default model parameters file in this repo was the one used to generate the numbers for the 500 test cases table, but that run was trained for the conference paper in 2018 before I knew about wandb so it is super hard to confirm that. There wasn't anything particularly special about that run, just that it was the one that was used for that table, whereas the 5 repeated runs were used for the rolling reward curve (and banging my head against the wall trying to re-create those results for the journal paper is why I started using wandb).

Somewhere on an old hard drive I believe/hope I have plots of the 500 test case trajectories for each number of agent, but I will keep digging to see if I can track that down. That wouldn't directly help you to run GA3C-CADRL in the future, but could help in debugging whether something has changed in the gym environment since those results were calculated.

(This is less relevant, but partially writing this down for my own future reference) There are also 5 corresponding `access_rnn_phase1?` runs -- to match up the phase 1 runs with the phase 2 runs, look at the RUN_ID used in the phase 2 `diff.patch` to the ID column of the phase 1 run (e.g., phase 2 run with`self.LOAD_FROM_WANDB_RUN_ID = 'run-20200402_205111-2pka2zpr'` corresponds to the phase 1 run with ID `2pka2zpr`).

**Comment by zhangqq21** [comment]
Ok, thank you for your reply!
